autogen/agentchat/conversable_agent.py

```diff
@@ -99,11 +100,14 @@ def __init__(
                 for available options.
                 To disable llm-based auto reply, set to False.
             default_auto_reply (str or dict or None): default auto reply when no code execution or llm-based reply is generated.
+            description (str): a short description of the agent. This description is used by other agents
+                (e.g. the GroupChatManager) to decide when to call upon this agent. (Default: system_message)
         """
         super().__init__(name)
         # a dictionary of conversations, default value is list
         self._oai_messages = defaultdict(list)
         self._oai_system_message = [{"content": system_message, "role": "system"}]
+        self.description = description if description is not None else system_message
```

Comment:
What if, starting from this PR, we clearly distinguish between system messages and agent descriptions, and no longer confuse system messages with agent descriptions?

Reply:
The idea here was to maintain backward compatibility. But ideally the idea of using system messages as default descriptions should be deprecated over time.